src/orc.cpp

```diff
@@ -67,41 +67,59 @@ std::string_view path_to_symbol(std::string_view path) {
 }
 
 /**************************************************************************************************/
+
 bool type_equivalent(const attribute& x, const attribute& y);
-dw::at find_attribute_conflict(const attribute_sequence& x, const attribute_sequence& y) {
-    ZoneScoped;
 
-    auto yfirst = y.begin();
-    auto ylast = y.end();
+bool attributes_conflict(dw::at name, const attribute& x, const attribute& y) {
+    if (name == dw::at::type && type_equivalent(x, y)) {
+        return false;
+    }
 
-    for (const auto& xattr : x) {
-        auto name = xattr._name;
-        if (nonfatal_attribute(name)) continue;
+    return x != y;
+}
 
-        auto yfound = std::find_if(yfirst, ylast, [&](auto& x) { return name == x._name; });
-        if (yfound == ylast) return name;
+std::vector<dw::at> fatal_attribute_names(const attribute_sequence& x) {
+    std::vector<dw::at> result;
+    for (const auto& entry : x) {
+        if (nonfatal_attribute(entry._name)) continue;
+        result.push_back(entry._name);
+    }
+    std::sort(result.begin(), result.end());
+    return result;
+}
 
-        const auto& yattr = *yfound;
+std::vector<dw::at> find_attribute_conflict(const attribute_sequence& x, const attribute_sequence& y) {
+    ZoneScoped;
 
-        if (name == dw::at::type && type_equivalent(xattr, yattr))
-            continue;
-        else if (xattr == yattr)
-            continue;
+    const auto x_names = fatal_attribute_names(x);
+    const auto y_names = fatal_attribute_names(y);
 
-        return name;
-    }
+    std::vector<dw::at> result;
+    std::vector<dw::at> intersection;
+
+    std::set_symmetric_difference(x_names.begin(), x_names.end(),
```

Comment:
Nice of of the 'set' stuff to clarify the code

Reply:
Thanks - I am much happier with this routine now than the state it was in before.